sympy/printing/fcode.py

```diff
@@ -106,6 +106,15 @@ def _get_statement(self, codestring):
 
     def _get_comment(self, text):
         return "! {0}".format(text)
+    
+    #issue 12267
+    def _print_sign(self,func):
+        if func.args[0].is_integer:
+            return "merge(0, isign(1, {0}), {0} == 0)".format(self._print(func.args[0]))
+        elif func.args[0].is_complex:
+            return "merge(cmplx(0d0, 0d0), {0}/abs({0}), abs({0}) == 0d0)".format(self._print(func.args[0]))
```

Comment:
`'      merge(cmplx(0d0, 0d0, kind(1d0)), x/abs(x), abs(x) == 0d0)'`
or
`'      merge(cmplx(0d0, 0d0, kind(x)), x/abs(x), abs(x) == 0d0)'`
which one is required